packages/@sanity/state-router/src/useRouterState.ts

```diff
@@ -1,15 +1,26 @@
-import {useContext, useEffect, useState} from 'react'
+import {useContext, useEffect, useState, useRef} from 'react'
 import {RouterState} from './components/types'
 import {RouterContext} from './RouterContext'
 
 export const useRouterState = (): RouterState | null => {
   const {channel, getState} = useContext(RouterContext)
   const [routerState, setState] = useState<RouterState>(getState())
 
+  // prevents "Can't perform a React state update on an unmounted component."
```

Comment:
Do you know why this happened? Can't see any obvious reason this should not work with the current code...

Reply:
it's possible that this could happen because there is a `setState` in the clean up portion of the original `useEffect`. I'm unsure it's possible that this could occur in app but in the tests definitely spin up and tear down multiple times